src/pydisagg/ihme/splitter/cat_splitter.py

```diff
@@ -0,0 +1,632 @@
+        List[str]
+            List of column names.
+        """
+        return list(set(self.index + [self.target, self.val, self.val_sd]))
+
+    @property
+    def val_fields(self) -> List[str]:
+        """
+        List of value fields (attributes).
+
+        Returns
+        -------
+        List[str]
+            List containing 'val' and 'val_sd'.
+        """
+        return ["val", "val_sd"]  # Attribute names
+
+
+class CatPatternConfig(Schema):
+    """
+    Configuration for the pattern DataFrame.
+
+    Parameters
+    ----------
+    index : List[str]
+        List of column names to be used as index in the pattern DataFrame.
+    target : str
+        Column name representing the target variable to split.
+    draws : List[str], optional
+        List of draw column names, by default [].
+    val : str, optional
+        Column name for the mean value in the pattern DataFrame, by default 'mean'.
+    val_sd : str, optional
+        Column name for the standard deviation in the pattern DataFrame, by default 'std_err'.
+    prefix : str, optional
+        Prefix to apply to column names when merging, by default 'cat_pat_'.
+    """
+
+    index: List[str]
+    target: str
+    draws: List[str] = []
+    val: str = "mean"
+    val_sd: str = "std_err"
+    prefix: str = "cat_pat_"
+
+    @property
+    def columns(self) -> List[str]:
+        """
+        List of all required columns in the pattern DataFrame.
+
+        Returns
+        -------
+        List[str]
+            List of column names.
+        """
+        return list(set(self.index + [self.target, self.val, self.val_sd]))
+
+    @property
+    def val_fields(self) -> List[str]:
+        """
+        List of value fields (attributes).
+
+        Returns
+        -------
+        List[str]
+            List containing 'val' and 'val_sd'.
+        """
+        return ["val", "val_sd"]  # Attribute names
+
+    def apply_prefix(self) -> dict:
+        """
+        Create a mapping to rename columns with the specified prefix.
+
+        Returns
+        -------
+        dict
+            Mapping from original column names to prefixed column names.
+        """
+        return {
+            self.val: f"{self.prefix}{self.val}",
+            self.val_sd: f"{self.prefix}{self.val_sd}",
+            self.target: self.target,  # Do not prefix the target column
+            **{idx: idx for idx in self.index},  # Keep index columns unchanged
+        }
+
+
+class CatPopulationConfig(Schema):
+    """
+    Configuration for the population DataFrame.
+
+    Parameters
+    ----------
+    index : List[str]
+        List of column names to be used as index in the population DataFrame.
+    target : str
+        Column name representing the target variable to split.
+    val : str
+        Column name for the population value.
+    prefix : str, optional
+        Prefix to apply to column names when merging, by default 'cat_pop_'.
+    """
+
+    index: List[str]
+    target: str
```

Comment:
For the population, I don't think we need `target`, like in the `age_splitter` we just need to make sure `parttern.target` is in the `index`